opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1904,12 +1921,37 @@ bool CpGrid::adapt()
     const std::vector<std::array<int,3>>& cells_per_dim_vec = {{2,2,2}}; // Arbitrary chosen values.
     std::vector<int> assignRefinedLevel(current_view_data_-> size(0));
     const auto& preAdaptMaxLevel = this ->maxLevel();
+    int local_marked_elem_count = 0;
     for (int elemIdx = 0; elemIdx < current_view_data_->size(0); ++elemIdx) {
         const auto& element = cpgrid::Entity<0>(*current_view_data_, elemIdx, true);
         assignRefinedLevel[elemIdx] = (this->getMark(element) == 1) ? (preAdaptMaxLevel +1) : 0;
+        if (this->getMark(element) == 1) {
+            ++local_marked_elem_count;
+        }
+    }
+    // Check if its a global refinement
+    bool is_global_refine = false;
+    std::vector<std::string> lgr_name_vec = { "LGR" + std::to_string(preAdaptMaxLevel +1) };
+    // Rewrite if global refinement
+#if HAVE_MPI
+    auto global_marked_elem_count = comm().sum(local_marked_elem_count);
+    auto global_cell_count_before_adapt = comm().sum(current_view_data_-> size(0)); // Recall overlap cells are also marked
+    if (global_marked_elem_count == global_cell_count_before_adapt) {
+        // GR stands for GLOBAL REFINEMET
+        lgr_name_vec = { "GR" + std::to_string(preAdaptMaxLevel +1) };
+        is_global_refine = true; // parallel
+    }
+#endif
+    if (local_marked_elem_count == current_view_data_-> size(0)) {
+        // GR stands for GLOBAL REFINEMET
+        lgr_name_vec = { "GR" + std::to_string(preAdaptMaxLevel +1) };
+        is_global_refine = true; // sequential
+    }
```

Comment:
Why do we need this check? If the user does call CpGrid::globalRefine() then it is a global refine. Otherwise I would not care.

Reply:
Good point! Somehow I thought it was needed to check that all cells - in all processes - got marked for refinement. It makes sense that if global refine has been invoked, then that's the case and the check is indeed unnecessary. Thanks!

Reply:
> Good point! But this piece of code is in adapt() definition, which is not necessary a global refinement. That's why the check is done.

It still seems like this could be done with an optional parameter to adapt that is only true when called from globalRefine.